plausible/client.go

```diff
@@ -63,6 +63,14 @@ func (c *Client) BaseURL() string {
 	return c.baseURL
 }
 
+func (c *Client) Event() *Event {
```

Comment:
Since `Event` is being built with information from the client I think we should do one of 2 things:

1) Make `Event` just take the pointer `c` to the plausible client.
2) Remove the need for `Event` and make `Push` be a function on `(c *Client)` and rename it to something like `PushEvent`.

I tend to prefer the 2nd alternative

Reply:
This is a great idea, I had a look at the rest of the code base again and tried to model it closer to that. So the code essentially calls `acquireRequest` on Event and then uses `doRequest()`.